Assets/Scripts/Controllers/HistoryController.cs

```diff
@@ -1,44 +1,235 @@
-﻿using Services.History;
+﻿using Model;
+using Services.History;
 using Services.History.Actions;
+using Services.RoutesFiles;
+using Services.SearchFiles;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
 using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
 
 namespace Controllers {
 	public class HistoryController : MonoBehaviour {
 		private NodeController nodeController;
-		private NetworkController networkController;
+		public NetworkController NetworkController { get; private set; }
 		private GraphController graphController;
 		public HistoryService HistoryService { get; private set; }
+		public GameObject RouteTemplate;
+		public GameObject RouteParent;
+		public GameObject SearchTemplateArticle;
+		public GameObject SearchTemplateCategory;
+		public GameObject SearchParent;
+		public int secondsToChangeRoute = 8;
+		public int numberOfDisplayedSearchEntries = 10;
+		public GameObject searchBox;
+		public GameObject SearchScrollView;
+		
+		public static Action startRouteAutoAction;
+		public static Action endRouteAutoAction;
```

Comment:
- why are they static?
- usual naming conversion would be `OnRouteStarted/Ended`
- should end up in `(Route)Controller` so they can be easily reached from other parts of application (actually had to move them myself when adding syncing for the AUTO icon)

Reply:
After seeing the action usage in Header it would be better to replace these two events with just one: `OnRoutePlayStateChanged<bool>`